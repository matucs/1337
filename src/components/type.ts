export interface IMember {
  name: string;
  email: string;
  phoneNumber: string;
  office: string;
  manager: string;
  orgUnit: string;
  mainText: string;
  gitHub: string;
  twitter: string;
  stackOverflow: string;
  linkedIn: string;
  imagePortraitUrl: string;
  imageWallOfLeetUrl: string;
  highlighted: boolean;
  published: boolean;
}

export type MemberState = {
  members: IMember[];
};

export type MemberAction = {
  type: string;
  member: IMember;
};

export type DispatchType = (args: MemberAction) => MemberAction;
