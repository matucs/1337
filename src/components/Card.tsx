import React from "react";

function Card() {
  return (
    <div className="card">
      <img src="assets/img/demopic/5.jpg" className="card-img-top" alt="..." />
      <div className="card-body">
        <div className="info">
          <h5 className="card-title">Card title</h5>
          <div className="social-media">
            <i className="icon fa fa-linkedin-square" aria-hidden="true"></i>
            <i className="icon fa fa-github" aria-hidden="true"></i>
            <i className="icon fa fa-twitter" aria-hidden="true"></i>
          </div>
        </div>
        <p className="card-text">Office: London</p>
      </div>
    </div>
  );
}

export default Card;
