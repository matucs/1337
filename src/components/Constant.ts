export const WEBSITE_URL = "https://tretton37.com/";
