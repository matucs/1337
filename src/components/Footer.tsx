import React from "react";

function Footer() {
  return (
    <div>
      <div className="footer">
        <p className="pull-left">Copyright &copy; 2022 13|37</p>
        <p className="pull-right">designed by Mehdi Bayat</p>
        <div className="clearfix"></div>
      </div>
    </div>
  );
}

export default Footer;
