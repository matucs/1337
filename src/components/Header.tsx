import React from "react";

function Header() {
  return (
    <div className="mainheading">
      <h1 className="sitetitle">13|37</h1>
      <p className="lead">Lista of our team members</p>
    </div>
  );
}

export default Header;
