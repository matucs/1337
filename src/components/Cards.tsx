import {  useSelector } from "react-redux";
import Card from "./Card";
import { IMember, MemberState } from "./type";

function Cards() {
  const result = useSelector<MemberState, IMember[]>((state) => state.result);
  const loading = useSelector<MemberState, boolean>((state) => state.loading);

  return (
    <div className="card-columns">
      {loading ? (
        <h1 className="loader"></h1>
      ) : (
        result.map((member, index) => {
          return <Card member={member} key={index} />;
        })
      )}
    </div>
  );
}

export default Cards;
