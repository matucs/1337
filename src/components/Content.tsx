import React, { Dispatch, useState } from "react";
import { useDispatch } from "react-redux";
import { fetchMembers, filterBy, sortBy } from "../store/ActionCreators";
import Members from "./Members";
import Footer from "./Footer";
import Header from "./Header";
import {
  StyledContainer,
  StyledFilter,
  StyledFilterByName,
  StyledFilterByOffice,
  StyledFooter,
  StyledSortBy,
} from "./Styled";

const Content: React.FC<{}> = () => {
  const dispatch: Dispatch<any> = useDispatch();
  React.useEffect(() => {
    dispatch(fetchMembers());
  }, [dispatch]);

  const [name, useName] = useState("");
  const [office, useOffice] = useState("");
  const HandleChange = (e: React.ChangeEvent<HTMLSelectElement>) => {
    const value = e.currentTarget.value === "name" ? "name" : "office";
    dispatch(sortBy(value));
    useName("");
    useOffice("");
  };
  const HandleNameFilter = (e: React.ChangeEvent<HTMLInputElement>) => {
    const type = e.currentTarget.name === "filter-name" ? "name" : "office";
    useName(e.target.value);
    dispatch(
      filterBy({
        type,
        payload: [],
        isLoading: false,
        filter: { name: e.target.value, office },
      })
    );
  };
  const HandleOfficeFilter = (e: React.ChangeEvent<HTMLInputElement>) => {
    const type = e.currentTarget.name === "filter-name" ? "name" : "office";
    useOffice(e.target.value);
    dispatch(
      filterBy({
        type,
        payload: [],
        isLoading: false,
        filter: { name, office: e.target.value.trim() },
      })
    );
  };
  return (
    <>
      <StyledContainer>
        <Header />
        <section>
          <StyledSortBy>
            <h3>Sort by:</h3>
            <select onChange={(e) => HandleChange(e)}>
              <option id="0" value={"name"}>
                name
              </option>
              <option id="1" value={"office"}>
                office
              </option>
            </select>
            <StyledFilter>
              <StyledFilterByName>
                <label htmlFor="filter-name">
                  <h3>Enter a name</h3>
                </label>
                <input
                  onChange={(e) => HandleNameFilter(e)}
                  name="filter-name"
                  value={name}
                  id="filter-name"
                />
              </StyledFilterByName>
              <StyledFilterByOffice>
                <label htmlFor="filter-office">
                  <h3>Enter office</h3>
                </label>
                <input
                  onChange={(e) => HandleOfficeFilter(e)}
                  name="filter-office"
                  id="filter-office"
                  value={office}
                />
              </StyledFilterByOffice>
            </StyledFilter>
          </StyledSortBy>
          <hr className="solid" />
          <Members />
        </section>
      </StyledContainer>
      <StyledFooter>
        <Footer />
      </StyledFooter>
    </>
  );
};

export default Content;
