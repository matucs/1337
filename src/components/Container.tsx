import React from "react";
import Cards from "./Cards";
import Footer from "./Footer";
import Header from "./Header";
import { StyledContainer, StyledFooter } from "./Style";

function Container() {
  return (
    <>
      <StyledContainer>
        <div className="container">
          <Header />
          <section >
            <div className="section-title">
              <h2>
                <span>team</span>
              </h2>
            </div>
            <Cards />
          </section>
        </div>
      </StyledContainer>
      <StyledFooter>
        <Footer />
      </StyledFooter>
    </>
  );
}

export default Container;
