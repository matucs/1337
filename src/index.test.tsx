import React from "react";
import Card from "./components/Card";
import { IMember } from "./components/type";
import renderer  from "react-test-renderer";

describe("Cards should be render", () => {
  const member: IMember = {
    name: "test-member",
    email: "example@1337.com",
    phoneNumber: "0192323434",
    office: "Lurd",
    manager: "manager-member",
    orgUnit: "test",
    mainText: "this is test",
    gitHub: "test@github",
    twitter: "@twiter",
    stackOverflow: "@stackOverflow",
    linkedIn: "@linkedin",
    imagePortraitUrl: "",
    imageWallOfLeetUrl: "",
    highlighted: false,
    published: false,
  };
  test("card renders snapshots, too", () => {
    const component = renderer.create(<Card member={member} />);
    let card = component.toJSON();
    expect(card).toMatchSnapshot();
  });
});
