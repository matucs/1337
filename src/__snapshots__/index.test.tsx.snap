// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Cards should be render Card changes the class when hovered 1`] = `
<div
  className="sc-eCImPb hZnmxO"
>
  <div
    className="lazyload-wrapper "
  >
    <div
      className="lazyload-placeholder"
      style={
        Object {
          "height": undefined,
        }
      }
    />
  </div>
</div>
`;

exports[`Cards should be render Card changes the class when hovered 2`] = `
<div
  className="sc-eCImPb hZnmxO"
>
  <div
    className="lazyload-wrapper "
  >
    <div
      className="lazyload-placeholder"
      style={
        Object {
          "height": undefined,
        }
      }
    />
  </div>
</div>
`;

exports[`Cards should be render card renders snapshots, too 1`] = `
<div
  className="sc-eCImPb hZnmxO"
>
  <div
    className="lazyload-wrapper "
  >
    <div
      className="lazyload-placeholder"
      style={
        Object {
          "height": undefined,
        }
      }
    />
  </div>
</div>
`;
